frame/im-online/src/lib.rs

```diff
@@ -219,6 +227,58 @@ where
 	pub validators_len: u32,
 }
 
+/// A type that is the same as [`OpaqueNetworkState`] but with [`Vec`] replaced with
+/// [`WeakBoundedVec<Limit>`] where Limit is the respective size limit
+#[derive(Clone, Eq, PartialEq, Encode, Decode, MaxEncodedLen)]
+#[codec(mel_bound(EncodingLimit: Get<u32>, AddressesLimit: Get<u32>))]
+pub struct BoundedOpaqueNetworkState<EncodingLimit, AddressesLimit>
```

Comment:
note that you use here `EncodingLimit` for both the limit of `OpaquePeerId` and the limit of `OpaqueMultiAddr`.
Maybe we should also differenciate both. or at least put in the doc that this limit should be the max of both limits.

Reply:
I have committed changes to address that by differentiating the limits.